src/rhsmlib/services/register.py

```diff
@@ -123,6 +123,11 @@ def register(
                 usage=usage,
                 jwt_token=jwt_token,
             )
+            # When new consumer is created, then close all existing connections
+            # to be able to recreate new one
+            cp_provider = inj.require(inj.CP_PROVIDER)
+            cp_provider.close_all_connections()
+
```

Comment:
good idea; shouldn't we do it rather in `UnregisterService.unregister()`?

Reply:
We should definitely do it here. Maybe we should close connections after unregistration too.

Reply:
I did some modification as suggested. Connections are closed after `unregister()`.